elastic-apm-agent/pom.xml

```diff
@@ -158,6 +158,23 @@
                             <goal>run</goal>
                         </goals>
                     </execution>
+                    <execution>
+                        <id>combine-sources</id>
+                        <phase>package</phase>
+                        <configuration>
+
+                            <target name="merge-sources" description="Merge source jars">
+                                <zip destfile="${project.basedir}/target/${project.artifactId}-${project.version}-sources.jar">
+                                    <zipgroupfileset dir="${project.basedir}/../apm-agent/target/" includes="apm-agent-*-sources.jar"/>
```

Comment:
Looks good to me, I do however have an idea on how to get rid of the direct `apm-agent` file based reference:
* use the `maven-dependency-plugin` execution in this pom above to explicitly copy the apm-agent sources JAR into the target folder (using the [copy goal](https://maven.apache.org/plugins/maven-dependency-plugin/copy-mojo.html) )
* use the copied sources-jar from the target folder instead of the file path `${project.basedir}/../apm-agent/target/`

What do you think?

Reply:
I tried that a slightly different config, it requires the dependency to be declared as such which then would be included in the class files. Then I tried to exclude the dependencies after including them, then they don't show in the sources. So if I then do that and then copy the source jar, then do the merge it might work - but sounds like a very complex addition

Reply:
Actually I think I even tried it and couldn't get the sources if the classes were excluded, and couldn't prevent the classes from being the final class jar if they weren't even when I tried to exclude them also using the shade target. It was a really painful week

Reply:
nice change, thank you! Much better